### ISSUE TITLE
Directional LightmapGI denoising exhibits artifacts on surfaces close to emissive surface

### ISSUE THREAD

**Issue by TidB** [godotengine/godot#86815]
### Tested versions

- Reproducible in v4.2.stable.official [46dc27791]
- Not reproducible in v4.1.stable.official [970459615]

### System information

Godot v4.2.stable - Windows 10.0.19045 - Vulkan (Forward+) - dedicated Radeon RX 580 Series (Advanced Micro Devices, Inc.; 31.0.21902.5) - 12th Gen Intel(R) Core(TM) i5-12400 (12 Threads)

### Issue description

I often place emissive meshes near other geometry (e.g. TV screens), which makes the following artifacts apparent:
![image](https://github.com/godotengine/godot/assets/6843686/dc622144-5854-449e-bfc5-05a6bbecce90)

The artifacts seem to arise from the denoiser step, as the un-denoised results (taken without the denoiser option) look similar enough:
![image](https://github.com/godotengine/godot/assets/6843686/85cdb822-2425-4276-bfb7-335bc40b6e28)

While the denoised results show the artifacting (or perhaps, the lack of reducing the noise?):
![image](https://github.com/godotengine/godot/assets/6843686/5e1e09a5-6fc1-4a02-99ac-754bd73a2952)

I've only taken the topmost part of the directional lightmap file, but the other parts exhibit similar artifacting. Even though the example images were taken with a non-normal-mapped mesh, the result with a normal map is the same (you can try it out by using `wall_mat` instead of `flat_mat` inside the MRP). 

This seems related to the introduction of the JNLM denoiser (#81659), as going back to OIDN inside the same 4.2 version produces this result:
![image](https://github.com/godotengine/godot/assets/6843686/2e9dfbfa-cb58-4170-998f-26777b470d45)

Moving the emissive mesh a bit away (usually 1 m distance is enough) makes the artifacts disappear:
![image](https://github.com/godotengine/godot/assets/6843686/f736564d-0b40-43cc-a01e-6a6a798250d2)
![image](https://github.com/godotengine/godot/assets/6843686/73d145ab-739c-4bf4-b252-8674b835052f)

### Steps to reproduce

1. Create two meshes set up for LightmapGI, one of them emissive
2. Position the emissive object close to the other mesh (less than 50 cm away, the closer the more noticeable)
3. Activate LightmapGI's `directional` option and bake lightmaps

### Minimal reproduction project (MRP)

[directional-lightmap.zip](https://github.com/godotengine/godot/files/13836744/directional-lightmap.zip)

The main scene `test-dir-denois` contains the other scenes `dir` and `non-dir`, both of which have their own LightmapGI instance.

**Comment by DarioSamo** [comment]
Yeah this is basically the edge case JNLM can't handle unless you increase the denoiser strength, as the algorithm will limit itself to only denoising a certain range of luminance. Of course doing that will denoise other stuff more aggressively so there is a trade-off.

If the strength isn't enough to solve it, I'd encourage you to use OIDN here as well as the included version was far too slow due to being single threaded, so you'll get much faster results from using the new executable approach.
